astropy/io/fits/hdu/hdulist.py

```diff
@@ -689,6 +698,13 @@ def writeto(self, fileobj, output_verify='exception', clobber=False,
             hdu._prewriteto(checksum=checksum)
             try:
                 hdu._writeto(hdulist._file)
+            except IOError, e:
+                free_space = get_free_space_in_dir(os.getcwd())
+                hdulist_size = np.sum(hdu.size for hdu in self)
+                if not fileobj.file_like and free_space < hdulist_size:
```

Comment:
I think right here you have the meaning of `file_like` backwards.  The naming is confusing, admittedly.  "file-like" here is using "like" in a strict sense--i.e. *definitely* not a file, but just a file-like object.